### ISSUE TITLE
yyin uninitalized

### ISSUE THREAD

**Issue by aaakhare** [westes/flex#365]
With options reentrant and prefix enabled, yyin is not
getting prefix. While with these options, yy_start, yyin,
yyout and yy_buffer_stack could be uninitialized.
On running, crash occurs. Crash was on yyin uninitialized
read. Version verified are 2.6.1 and 2.6.4

%option reentrant
%option prefix

                if ( ! yyg->yy_start )
                        yyg->yy_start = 1;      /* first start state */

                if ( ! yyin )
                        yyin = stdin;

                if ( ! yyout )
                        yyout = stdout;

static void yyensure_buffer_stack (yyscan_t yyscanner)
{
        yy_size_t num_to_alloc;
    struct yyguts_t * yyg = (struct yyguts_t*)yyscanner;

        if (!yyg->yy_buffer_stack) {

**Comment by jannick0** [comment]
For *reentrant* scanners call `yylex` like so 
~~~c
yyscan_t lexer;
yylex_init( &lexer );
yylex( lexer );
yylex_destroy( lexer );
~~~
to  ensure that all objects required by `yylex` are initialized and destroyed afterwards. This works with `%option prefix="..."` as well. Does this help?

**Comment by aaakhare** [comment]
This init sequence is already done. In yy_init_globals, initialization is
done.
Still, on gdb, yyin was showing junk value. yyin and <prefix>yyin is still
the problem.
In generated scanner, <prefix>yyin doesn't show.

**Comment by Explorer09** [comment]
@aaakhare Can you please give us a proper bug report, including steps to reproduce (attach example files if necessary), actual result, and expected result?
Your information looks fragmented and it's difficult for us to help.

**Comment by jannick0** [comment]
You - and `gdb`- should not expect `yyin` to be a variable in any _reentrant_ scanner, since a preprocessor macro defines it to be a member of the scanner object which is initialized by `yylex_init` and passed on to `yylex`. If you want to appropriately access `yyin`, you may want to use the functions `yyget_in` and `yyset_in` (or their cousin functions if a prefix other than `yy` is used).

Having said that rather by guessing what your problem is about, it is difficult to help you more without any **minimal** stand-alone example producing the crash as you claim.

**Comment by aaakhare** [comment]
That is correct. in reentrant case, yyin has to be accessed through api.
Please close the issue.

**Comment by jannick0** [comment]
Great that the issue is solved. 

Could you as the originator of this issue please close the issue by either sending in an email containing the text "Closes: #365" (cf. https://help.github.com/articles/closing-issues-using-keywords) or by hitting the close button in the github web interface? Thanks.
